plugins/out_s3/s3.c

```diff
@@ -1350,16 +1562,14 @@ static void cb_s3_flush(const void *data, size_t bytes,
                             struct flb_config *config)
 {
```

Comment:
it not a best practice to create a func with too many lines and this will make it hard to maintain in the future. 
Could you refactor the cb_s3_flush to separate with several func?